Sources/InfluxDBSwift/Point.swift

```diff
@@ -163,6 +125,169 @@ extension InfluxDBClient {
     }
 }
 
+extension InfluxDBClient.Point {
+    /// Possible value types of Field
+    public enum FieldValue {
+        /// Support for Int8
+        init(_ value: Int8) {
+            self = .int(Int(value))
+        }
+        /// Support for Int16
+        init(_ value: Int16) {
+            self = .int(Int(value))
+        }
+        /// Support for Int32
+        init(_ value: Int32) {
+            self = .int(Int(value))
+        }
+        /// Support for Int64
+        init(_ value: Int64) {
+            self = .int(Int(value))
+        }
+        /// Support for UInt8
+        init(_ value: UInt8) {
+            self = .uint(UInt(value))
+        }
+        /// Support for UInt16
+        init(_ value: UInt16) {
+            self = .uint(UInt(value))
+        }
+        /// Support for UInt32
+        init(_ value: UInt32) {
+            self = .uint(UInt(value))
+        }
+        /// Support for UInt64
+        init(_ value: UInt64) {
+            self = .uint(UInt(value))
+        }
+        /// Support for Float
+        init(_ value: Float) {
+            self = .double(Double(value))
+        }
+
+        /// signed integer number
+        case int(Int)
+        /// unsigned integer number
+        case uint(UInt)
+        /// floating number
+        case double(Double)
+        /// true or false value
+        case boolean(Bool)
+        /// string value
+        case string(String)
+    }
+
+    /// Possible value types of Field
+    public enum TimestampValue: CustomStringConvertible {
+        // The number of ticks since the UNIX epoch. The value has to be specified with correct precision.
+        case interval(Int, InfluxDBClient.TimestampPrecision = InfluxDBClient.defaultTimestampPrecision)
+        // The date timestamp.
+        case date(Date, InfluxDBClient.TimestampPrecision = InfluxDBClient.defaultTimestampPrecision)
```

Comment:
`date` does not require a `TimestampPrecision`, as the precision is fixed based on the `Date` data structure.

Reply:
I think it is same as for `interval`. The following constructions is valid and has to lead to two batches (different precision):

```swift
let pointFromSource1 = InfluxDBClient.Point("mem")
        .addTag(key: "tag", value: "a")
        .addField(key: "value", value: .int(1))
        .time(time: .date(parsedDate1, .s))

let pointFromSource2 = InfluxDBClient.Point("mem")
        .addTag(key: "tag", value: "b")
        .addField(key: "value", value: .int(2))
        .time(time: .date(parsedDate2, .ns))

client.makeWriteAPI().writeRecords(records: [pointFromSource1, [pointFromSource2]]) { _, _ in

}
```